Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,246 @@
+                            RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Flat;
+                            break;
+                    }
+                }
+                else
+                {
+                    RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Skybox;
+                }
+            }
+        }
+        
+        void UpdateLight()
+        {
+            Quaternion tilt = Quaternion.AngleAxis(sunTilt, Vector3.forward);
+            Quaternion rot = Quaternion.AngleAxis((CurrentTime01 * 360) - 90, Vector3.right);
+
+            directionalLight.transform.rotation = tilt * rot; // Yes axial tilt
+            directionalLight.transform.Rotate(Vector3.up, sunRotationOffset - 90, Space.World);
+            moon.transform.forward = -directionalLight.transform.forward;
+
+            if (timecycle)
+            {
+                // Sun & moon's color and brightness
+                if (timecycle.overrideSunlight)
+                {
+                    directionalLight.color = timecycle.sunlightColor.Evaluate(CurrentTime01);
+                    directionalLight.intensity = timecycle.sunlightColor.Evaluate(CurrentTime01).a * timecycle.maxSunlightIntensity;
+                }
+                if (timecycle.overrideMoonlight)
+                {
+                    Light moonLight = moon.GetComponent<Light>();
+                    moonLight.color = timecycle.moonlightColor.Evaluate(CurrentTime01);
+                    moonLight.intensity = timecycle.moonlightColor.Evaluate(CurrentTime01).a * timecycle.maxMoonlightIntensity;
+                }
+
+                // Environment lighting
+                if (timecycle.overrideEnvironmentLighting)
+                {
+                    switch (timecycle.lightingOverrideMode)
+                    {
+                        case TimecycleProfile.AmbientLightingOverrideMode.Gradient:
+                            RenderSettings.ambientSkyColor = timecycle.skyColor.Evaluate(CurrentTime01);
+                            RenderSettings.ambientEquatorColor = timecycle.equatorColor.Evaluate(CurrentTime01);
+                            RenderSettings.ambientGroundColor = timecycle.groundColor.Evaluate(CurrentTime01);
+                            break;
+                        case TimecycleProfile.AmbientLightingOverrideMode.Color:
+                            RenderSettings.ambientLight = timecycle.lightingColor.Evaluate(CurrentTime01);
+                            break;
+                    }
+                }
+
+                // Fog
+                if (timecycle.overrideFog)
+                {
+                    RenderSettings.fogColor = timecycle.fogColor.Evaluate(CurrentTime01);
+                    RenderSettings.fogDensity = timecycle.fogColor.Evaluate(CurrentTime01).a * timecycle.maxFogDensity;
+                }
+            }
+        }
+
+        void OnValidate()
```

Comment:
"This function is called when the script is loaded or a value is changed in the Inspector (Called in the editor only)."

So, since this can only be called by the editor, where is there an extra #if UNITY_EDITOR region?

Reply:
@WattanaGaming Your addition as well, please check it out if you have time to.